resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/hadooputils/HadoopBootstrapUtil.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.features.hadooputils
+
+import java.io.File
+
+import scala.collection.JavaConverters._
+
+import com.google.common.base.Charsets
+import com.google.common.io.Files
+import io.fabric8.kubernetes.api.model._
+
+import org.apache.spark.SparkException
+import org.apache.spark.deploy.k8s.Constants._
+import org.apache.spark.deploy.k8s.SparkPod
+
+private[spark] object HadoopBootstrapUtil {
+
+   /**
+    * Mounting the DT secret for both the Driver and the executors
+    *
+    * @param dtSecretName Name of the secret that stores the Delegation Token
+    * @param dtSecretItemKey Name of the Item Key storing the Delegation Token
+    * @param userName Name of the SparkUser to set SPARK_USER
+    * @param maybeFileLocation Optional Location of the krb5 file
+    * @param newKrb5ConfName Optional location of the ConfigMap for Krb5
+    * @param maybeKrb5ConfName Optional name of ConfigMap for Krb5
+    * @param pod Input pod to be appended to
+    * @return a modified SparkPod
+    */
+  def bootstrapKerberosPod(
+    dtSecretName: String,
+    dtSecretItemKey: String,
+    userName: String,
+    maybeFileLocation: Option[String],
```

Comment:
You don't need `maybeFileLocation` given the `newKrb5ConfName` below. BTW: `newKrb5ConfName` should be `Option[String] instead.`